lib/storage/keyval/etcd.go

```diff
@@ -101,6 +104,30 @@ func LocalEtcdConfig(retryTimeout time.Duration) (*ETCDConfig, error) {
 	}, nil
 }
 
+// EtcdBackend enables access to etcd-specific features
+type EtcdBackend interface {
```

Comment:
This is a good point. I don't think it would be a downside to let the reads go through fsm - etcd3 is doing that by default. Let me make this a default and run a couple of tests - I suspect I won't notice a thing.

Reply:
It does bring a bit of overhead - the tests took additional ~30min.

Reply:
Moving the backend to etcdv3... yea... that's a decent amount of work no one has time for, it would be great to overhaul. What I suspect is more likely, on the next version of etcd when it's no longer considered experimental, is to use the v2 API with the v3 backend.